_docs/Qabel-Protocol-Box.md

```diff
@@ -0,0 +1,578 @@
+},
+```
+
+### File Metadata [FM]
+
+A file only has its own FM, if it is shared in a single file share.
+The FM is stored in the path referenced as "meta" in the file object and encrypted
+with a new symmetric key
+
+```
+{
+name: STR, // filename
+spec_version: INT,  // version of the VOLUME spec
+size: LONG, // uncompressed file size
+mtime: LONG, // modification time as seconds since epoch
+key: KEY, // symmetric key for the block
+block: STR // path to the block without the prefix \<root\>/blocks/
+}
+```
+
+### Objects
+
+File:
+
+```
+{
+name: STR, // object name,
+size: LONG, // uncompressed file size
+mtime: LONG, // modification time as seconds since epoch
+meta: {ref: STR, // path to the FM, if it exists
+       key: KEY} // symmetric key for the FM
+key: KEY, // symmetric key for the block
+block: STR // path to the block without the prefix \<root\>/blocks/
+},
+```
+
+Folder:
+
+```
+{
+name: STR // object name,
+key: KEY // symmetric directory key
+ref: STR // ref of the metadata file that contains information about the folder
+},
+```
+
+External:
+
+```
+{
+name: STR, // object name,
+key: KEY // symmetric directory key
+owner: STR, // public key of the owner of that VOLUME
+url: STR // URL to the metadata file that contains information about the folder
+},
+```
+
+### dk - Directory Key
+The directory key is stored in the directory object of the parent folder, the index
+file DM is encrypted with the public key of the owner.
+
+### fk - File Key
+File keys are stored in the directory metadata file
+
+### Qabel Identities
+Identities have a public key **pub** and a private key **priv**
+
+### Device ID
+Each client device has a unique ID which is a random generated UUID **devID**
+
+### Path to VOLUME/\<index\>
+\<index\> is calculated from the sha256 of the owners private key. Take the first 128bit of it
+and format them like the canonical form of a UUID.
+This means: 8-4-4-4-12 hexadecimal digits.
+Example: e5cceedc-c222-d549-6211-1b6c684e0b2a
+
+### Quota tracking
+Quota tracking is done by AWS. The S3 service is configured to call an AWS Lambda method for each
+s3:ObjectCreated and s3:ObjectRemoved in VOLUME/blocks. The Lambda method keeps track of the quota
+by incrementing or decrementing the amount of used space saved in DynamoDB. The accounting server
+then regularly requests all quota data and inserts it into his database. Clients can request
+this information for their VOLUME by calling a REST method on the accounting server.
+
+
+### Share notification drop message
+The plaintext of the drop message is a JSON document with the url of the DM and the symmetric key.
+
+```
+{
+	url: STR // url to the DM of the shared folder
+	key: KEY // symmetric directory key
+}
+```
+
+## Initializing a new VOLUME
+
+### Task
+
+Initialize a new VOLUME without any objects
+
+### Prerequisites
+
+* Valid federation token with write access to the VOLUME
+* Device ID **devId0*
```

Comment:
If you want to document the device ID as a prerequisite, you should add it to many of the following tasks to, because they update the DM version, too.